### PR TITLE
doc: backup operator design doc

### PR DESCRIPTION
**Pull request by fanminshi**
[skip ci]

### CODE REVIEW

File: doc/design/backup_operator.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  storageType: s3
+  backupIntervalInSecond: 30
+  s3:
+    s3Bucket: <your-bucket>
+    prefix: <your-prefix>
+    awsSecret: <your-secret>
+```
+
+* Creates backup sidecar with appropriate resource such as s3 secrets.
+* Spawns the sidecar pod.
```
Comment (xiang90) [inline doc/design/backup_operator.md]: sidecar pod is not really needed. we can just do the backup in process for simplicity.

File: doc/design/backup_operator.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    kind: EtcdBackup
+    plural: etcdbackups
+```
+
+* Create an informer that listens on the backup CRD
+Create/manage a backup sidecar pod for each new CR:
+For instance, the new CR for S3 backup:
+
+```yaml
+apiVersion: "etcd.database.coreos.com/v1alpha1"
```
Comment (xiang90) [inline doc/design/backup_operator.md]: a backup should have a status field to indicate if the backup is created or failed, etc..

File: doc/design/backup_operator.md (hunk 1 above)
Comment (fanminshi) [reply doc/design/backup_operator.md]: yeah, that's much simpler.

File: doc/design/backup_operator.md (hunk 2 above)
Comment (hongchaodeng) [reply doc/design/backup_operator.md]: To keep it aligned with etcd operator's group version:
```
.../v1beta2
```

**Comment by fanminshi** [comment]
all fixed. PTAL

**Comment by fanminshi** [comment]
TODO: add status field

File: doc/design/backup_operator.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  version: v1alpha1
+  Scope: Namespaced
+  names: 
+    kind: EtcdBackup
+    plural: etcdbackups
+```
+
+CR Definition:
+```yaml
+apiVersion: "etcd.database.coreos.com/v1alpha1"
```
Comment (hongchaodeng) [inline doc/design/backup_operator.md]: v1beta2

File: doc/design/backup_operator.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+kind: "EtcdBackup"
+metadata:
+  name: example-etcd-cluster
```
Comment (hongchaodeng) [inline doc/design/backup_operator.md]: example-etcd-cluster-backup

File: doc/design/backup_operator.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+spec:
+  clusterName: example-etcd-cluster
+  storageType: <your-storage-type>
+  <your-storage-type>:
```
Comment (hongchaodeng) [inline doc/design/backup_operator.md]: If it is not adding any details here, probably just use `...` for everything from below.

Then start defining Go type for BackupSpec:
```Go
type BackupSpec struct {
}
... Status
```

**Comment by fanminshi** [comment]
@hongchaodeng updated the doc. ptal.

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**